tests/nested/manual/uc-grub-boot-chains/task.yaml

```diff
@@ -0,0 +1,95 @@
+  # snappy-dev:image ppa is possibly enabled by some other tests.
+  cat >/etc/apt/preferences.d/ppa-disable-shim-signed <<EOF
+  Package: shim-signed
+  Pin: release o=LP-PPA-snappy-dev-image
+  Pin-Priority: -1
+  EOF
+  apt update
+
+  VERSION=$(tests.nested show version)
+  snap download --basename=pc --channel="${VERSION}/${NESTED_GADGET_CHANNEL}" pc
+  unsquashfs -d pc pc.snap
+  apt download shim-signed
+  dpkg-deb -x shim-signed_*.deb ./shim-signed
+  shim="${PWD}/shim-signed/usr/lib/shim/shimx64.efi.dualsigned"
+  fb="${PWD}/shim-signed/usr/lib/shim/fbx64.efi"
+  csv="${PWD}/shim-signed/usr/lib/shim/BOOTX64.CSV"
+
+  KEY_NAME=$(tests.nested download snakeoil-key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+  python3 modify-gadget.py 0 removable pc/meta/gadget.yaml gadget.yaml.old
+  python3 modify-gadget.py 1 boot_entry pc/meta/gadget.yaml gadget.yaml.new
+  python3 modify-gadget.py 2 removable pc/meta/gadget.yaml gadget.yaml.old-again
+
+  cp "${shim}" pc/shim.efi.signed
+  rm -f pc/boot.csv
+  rm -f pc/fb.efi
+  cp gadget.yaml.old pc/meta/gadget.yaml
+
+  tests.nested secboot-sign gadget pc "${SNAKEOIL_KEY}" "${SNAKEOIL_CERT}"
+  snap pack pc "$(tests.nested get extra-snaps-path)"
+
+  cp "${fb}" pc/fb.efi
+  cp "${csv}" pc/boot.csv
+  cp gadget.yaml.new pc/meta/gadget.yaml
+  tests.nested secboot-sign gadget pc "${SNAKEOIL_KEY}" "${SNAKEOIL_CERT}"
+  snap pack pc --filename=pc_x2.snap
+
+  rm -f pc/fb.efi
+  rm -f pc/boot.csv
+  cp gadget.yaml.old-again pc/meta/gadget.yaml
+  tests.nested secboot-sign gadget pc "${SNAKEOIL_KEY}" "${SNAKEOIL_CERT}"
+  snap pack pc --filename=pc_x3.snap
+
+  tests.nested build-image core
+  tests.nested create-vm core
+
+restore: |
+  rm -f /etc/apt/preferences.d/ppa-disable-shim-signed
+  apt update
+
+execute: |
+  remote.exec "! [ -f /boot/efi/EFI/ubuntu/grubx64.efi ]"
+  remote.exec "[ -f /boot/efi/EFI/boot/grubx64.efi ]"
+  remote.push pc_x2.snap
+  boot_id=$(tests.nested boot-id)
+  remote.exec sudo snap install --dangerous pc_x2.snap
+  remote.wait-for reboot "$boot_id"
+  remote.exec "[ -f /boot/efi/EFI/ubuntu/grubx64.efi ]"
+  remote.exec "[ -f /boot/efi/EFI/ubuntu/shimx64.efi ]"
+  remote.exec "[ -f /boot/efi/EFI/ubuntu/bootx64.csv ]"
+
+  #FIXME: We do not remove old assets yet
+  #remote.exec "! [ -f /boot/efi/EFI/boot/grubx64.efi ]"
+  remote.exec "sudo rm -f /boot/efi/EFI/boot/grubx64.efi"
+
+  # Make sure we can still boot
+  boot_id=$(tests.nested boot-id)
+  remote.exec sudo reboot || true
+  remote.wait-for reboot "$boot_id"
+
+  remote.exec "! [ -f /boot/efi/EFI/boot/grubx64.efi ]"
+
+  # Let's "downgrade"!
+  #remote.push pc_x3.snap
+  #boot_id=$(tests.nested boot-id)
+  #remote.exec sudo snap install --dangerous pc_x3.snap
+  #remote.wait-for reboot "$boot_id"
+  #remote.exec "[ -f /boot/efi/EFI/boot/grubx64.efi ]"
+  #retry --wait 4 -n 100 sh -c 'remote.exec "! [ -f /boot/efi/EFI/ubuntu/grubx64.efi ]"'
```

Comment:
is this not supported? will it be?

Reply:
Oh! I am missing a fixme here.

I need to fix the deletion of gadget content that is not available in the next gadget. In case of downgrade, the "next gadget"  should not have "fbx64.efi", so we should remove it, because it changes the behavior of shim.

I will do this fix in a different PR. But I will for sure comment here on why this does not work. And why it needs to work.